components/omega/src/ocn/VertMix.h

```diff
@@ -0,0 +1,210 @@
+
+   // Convective mixing parameters
+   Real ConvDiff =
+       1.0; ///< Convective vertical viscosity and diffusivity (m^2 s^-1)
+   Real ConvTriggerBVF = 0.0; ///< Reference density (kg m^-3) at (T,S)=(0,0)
+
+   /// Constructor for ConvectiveMix
+   ConvectiveMix(const VertCoord *VCoord);
+
+   KOKKOS_FUNCTION void operator()(Array2DReal VertDiff, Array2DReal VertVisc,
+                                   I4 ICell,
+                                   const Array2DReal &BruntVaisalaFreq) const {
+      for (int K = 1; K < NVertLayers; ++K) {
+         if (K == 0) {
+            VertVisc(ICell, K) = 0.0_Real;
+            VertDiff(ICell, K) = 0.0_Real;
+         } else {
+            if (BruntVaisalaFreq(ICell, K) < ConvTriggerBVF) {
+               VertDiff(ICell, K) = VertDiff(ICell, K) + ConvDiff;
+               VertVisc(ICell, K) = VertVisc(ICell, K) + ConvDiff;
+            }
+         }
+      }
+   }
+
+ private:
+   I4 NVertLayers;
+};
+
+class PPShearMix {
+ public:
+   bool Enabled = true; ///< Enable shear mixing flag
+
+   // Shear mixing parameters
+   Real ShearNuZero =
+       0.005; ///< Numerator of Pacanowski and Philander (1981) Eq (1).
+   Real ShearAlpha = 5.0; ///< Alpha value used in Pacanowski and Philander
+                          ///< (1981) Eqs (1) and (2).
+   Real ShearExponent =
+       2.0; /// Exponent value used in Pacanowski and Philander (1981) Eqs (1).
+
+   /// Constructor for PPShearMix
+   PPShearMix(const HorzMesh *Mesh, const VertCoord *VCoord);
+
+   KOKKOS_FUNCTION void operator()(Array2DReal VertDiff, Array2DReal VertVisc,
+                                   I4 ICell, const Array2DReal &NormalVelocity,
+                                   const Array2DReal &TangentialVelocity,
+                                   const Array2DReal &BruntVaisalaFreq) const {
+
+      for (int K = 0; K < NVertLayers; ++K) {
+         if (K == 0) {
+            VertVisc(ICell, K) = 0.0_Real;
+            VertDiff(ICell, K) = 0.0_Real;
+         } else {
+            Real ShearViscVal = 0.0;
+            Real InvAreaCell  = 1.0 / AreaCell(ICell);
+            Real ShearSquared = 0.0;
+            for (int J = 0; J < NEdgesOnCell(ICell); ++J) {
+               I4 JEdge = EdgesOnCell(ICell, J);
+               Real Factor =
+                   0.5_Real * DcEdge(JEdge) * DvEdge(JEdge) * InvAreaCell;
+               Real DelU2 = Kokkos::pow(NormalVelocity(JEdge, K - 1) -
+                                            NormalVelocity(JEdge, K),
+                                        2.0_Real) +
+                            Kokkos::pow(TangentialVelocity(JEdge, K - 1) -
+                                            TangentialVelocity(JEdge, K),
+                                        2.0_Real);
+               ShearSquared = ShearSquared + Factor * DelU2;
+            }
+            ShearSquared =
+                ShearSquared /
+                Kokkos::pow(ZMid(ICell, K - 1) - ZMid(ICell, K), 2.0_Real);
```

Comment:
Don't know how efficient Kokkos::pow is, but often for low integral powers (square, cube), actually doing the multiply is faster than using the pow function.